src/depcache.erl

```diff
@@ -120,19 +134,85 @@ start_link(Config) ->
 %% [http://erlang.org/doc/man/gen_server.html#start_link-4 gen_server:start_link/4].
 %%
 %% For Config, you can pass:
-%% `{callback, {Module, Function, Arguments}}': depcache event callback
-%% `{memory_max, MaxMemoryInMB}': number of MB to limit depcache size at.
+%% <dl>
+%%   <dt>`callback => {Module, Function, Arguments}'</dt>
+%%   <dd>depcache event callback</dd>
+%%   <dt>`memory_max => MaxMemoryInMB'</dt>
+%%   <dd>number of MB to limit depcache size at</dd>
+%% </dl>
 %% @param Name of process
 %% @param Config configuration options
 %% @returns Result of starting gen_server item.
 
 -spec start_link(Name, Config) -> Result when 
     Name :: atom(), 
-    Config :: proplist(),
+    Config :: config(),
     Result :: {ok, pid()} | ignore | {error, term()}.
+start_link(Name, Config) when is_map(Config) ->
+    ensure_valid_config(Config),
+    gen_server:start_link({local, Name}, ?MODULE, Config#{name => Name}, []);
 start_link(Name, Config) ->
-    gen_server:start_link({local, Name}, ?MODULE, [{name,Name}|Config], []).
+    start_link(Name, proplists_to_map(Config)).
 
+-ifdef(OTP_RELEASE).
+    -if(?OTP_RELEASE >= 24).
+        -define(has_proplists_to_map, true).
+    -endif.
+-endif.
+
+-spec proplists_to_map(PropList) -> Map when
+    PropList :: proplist(),
+    Map :: map().
+-ifdef(has_proplists_to_map).
+%% proplists:to_map was introduced in OTP 24.
+proplists_to_map(List) ->
+    proplists:to_map(List).
+-else.
+%% Backport of proplists:to_map.
+%% Copied from https://github.com/erlang/otp/blob/master/lib/stdlib/src/proplists.erl#L704-L720
+proplists_to_map(List) ->
+    lists:foldr(
+        fun
+            ({K, V}, M) ->
+                M#{K => V};
+            %% if tuples with arity /= 2 appear before atoms or
+            %% tuples with arity == 2, proplists:get_value/2,3 returns early
+            (T, M) when 1 =< tuple_size(T) ->
```

Comment:
Shouldn't this have an extra guard `is_tuple(T)` ?

As `tuple_size(some_atom)` gives a `badarg` exception, which makes it impossible to reach the next clause where there is a test on `is_atom(K)`

Reply:
Which leaves me wondering why the `proplists:to_map/1` function doesn't error on this...

Reply:
Guards don't error, or rather, an error in a guard just makes the guard fail. So with `tuple_size` in a guard, you get `is_tuple` included with it for free ;)

(If you wonder why this gymnastic is needed to turn a proplist into a map (instead of just using `maps:from_list/1`), see [here](https://github.com/erlang/otp/pull/2910) and especially [here](https://github.com/erlang/otp/pull/2910#issuecomment-743090874).)

Reply:
Everyday I learn something new 😄 

So I can hard crash my guard, just to not let me pass through. A failed and a crashed guard are both assumed to block the door, I guess.

Reply:
I think there was even a topic on the forum circling around that lately... Ah yes, here: https://erlangforums.com/t/guard-expressions-raising-errors/1668